src/auth/webhook-secret.guard.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import * as crypto from 'crypto';
+import { Request } from 'express';
+import { Env } from 'src/env';
+
+@Injectable()
+export class WebhookSecretGuard implements CanActivate {
+  canActivate(context: ExecutionContext): boolean {
+    const request: Request = context.switchToHttp().getRequest();
+    const signature = request.headers['x-hub-signature-256'] as string;
+    const secret = Env.SECRET.WEBHOOK;
```

Comment:
PullProofAI

Consider adding null checks for `request`, `request.headers`, and `request.body` to prevent potential runtime errors if any of these are undefined.
```diff
+     if (!request || !request.headers || !request.body) {
+       throw new UnauthorizedException('Invalid request');
+     }
```



<!-- This is an auto-generated comment by PullProofAI -->